experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -328,6 +328,10 @@ export class LazySequence<TTypes extends AllowedTypes>
 		sourceEnd: number,
 		source?: Sequence<CheckTypesOverlap<TTypesSource, TTypes>>,
 	): void {
+		if (source !== undefined) {
```

Comment:
I think the intent of the original assert was to use `sourceField` and not `source`, as the next line after it casts it to a LazySequence. Which do you think is better?

Reply:
Checking `sourceField` seems ok but still feels better to move the check further up to avoid unnecessary computation if we know we're going to fail. The new commit goes that route and uses a type predicate to avoid the `as` cast. Seeing the `instanceof` leaves me a bit concerned, with the issues it has between package versions, but this at least shouldn't introduce new failure modes.